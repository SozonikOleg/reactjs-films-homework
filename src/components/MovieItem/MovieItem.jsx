
import React, { Component } from 'react';
import styles from './MovieItem.module.scss';
import Modal from '../Modal/Modal'
import { connect } from 'react-redux';
import { getDataTrailer } from '../../action/getDataTrailer';
import { BrowserRouter as Router, Route, Link } from 'react-router-dom';

class MovieItem extends Component {
  constructor(props) {
    super(props);
    this.state = {
      itemState: true,
      discriptionState: false,
      isModalOpen: false
    };
  }

  toggleState() {
    this.setState({ isOpepned: !this.state.isOpepned });
  }

  ClickViewButton() {
    this.setState({ isOpepned: !this.state.isOpepned });
    this.setState({ discriptionState: true });
    this.setState({ itemState: false });
  }

  closeDiscription() {
    this.setState({ discriptionState: false });
    this.setState({ itemState: true });
  }

  toggleModal = () => {
    this.setState(state => ({isModalOpen: !state.isModalOpen}));
    this.props.returnId(this.props.id);
    this.setState({ discriptionState: false })
    this.setState({ itemState: true })
  };

  render() {
    const url = `https://image.tmdb.org/t/p/w400${this.props.background}`;
    const dataItem = this.props.dataItem;
    
    let itemState;

    if(this.state.itemState){
      itemState = (
        <section 
          className={styles.movie_item}
        >
          <div className={styles.background_img}>
            <img src={url} className={styles.movie_img} />
            <div className={styles.movie_hover}>
              <button onClick={this.toggleModal} className={styles.hover_button_play} />
              <span className={styles.hover_watch}>Watch Now</span>
              <button onClick={this.ClickViewButton.bind(this)} className={styles.hover_button_view}>View Info</button>
            </div>
          </div>

          <div className={styles.movie_description}>
            <div className="movie-info">
              <h2 className={styles.title}>{this.props.itemData}</h2>
              <ul className={styles.movie_genre}>
                <li>Action</li>
                <li>Drama</li>
                <li>Comedy</li>
              </ul>
            </div>
            <div className={styles.movie_rating}>{this.props.vote_average}</div>
          </div>
        </section>
      )
    }

    let discriptionState;

    if (this.state.discriptionState) {
      discriptionState = (
        <div className={styles.discription_item}>
          <img src={url} alt="" className={styles.description_img} />
          <button className={styles.description_button} onClick={this.closeDiscription.bind(this)}></button>
          <div  className={styles.description_info} >
            <section className={styles.description_header}>
              <div className={styles.description_nav}>
                <h2>{dataItem.original_title}</h2>
                <ul className={styles.description_list}>
                  <li>Action</li>
                  <li>Adventure</li>
                  <li>Fantasy</li>
                </ul>
              </div>
              <div className={styles.vote_average}>{dataItem.vote_average}</div>
            </section>
            <section className={styles.main}>
              <p className={styles.discription_overview}>{dataItem.overview}</p>
            </section>
            <Link  to={{ pathname: location.search, search: `+nameId:${this.props.id}`,}}>
              <button onClick={this.toggleModal} className={styles.button_watch} >Watch Now</button>
            </Link>
          </div>
        </div>

      );
    }

    return (
      <Router>
        <div className={styles.movie_wrapper}>
          {itemState}
            {discriptionState}
            {this.state.isModalOpen &&
                <Modal id={this.props.id} onClose={this.toggleModal}>
                </Modal> 
            }
        </div>
      </Router>
    );
  }
}

export default connect(
  state => ({}),
  dispatch => ({
    returnId: id => {
      dispatch(getDataTrailer(id));
    },
  }),
)(MovieItem);
