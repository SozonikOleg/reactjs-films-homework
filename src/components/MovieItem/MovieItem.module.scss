.movie_item {
    height: 550px;
    color: #09ade3;
    position: relative;
    .movie_hover {
        z-index: 2;
    }
    .movie_img {
        height: 550px;
        width: 100%;
        position: relative;
        text-align: center;
    }
    .background_img {
        background: black;
        text-align: -webkit-center;
    }
}

.movie_wrapper {
    width:22.1%;
    margin: 27px;
    background-color: white;
}

// ************start hover movie item****************

.movie_hover {
    display: none;
    width: 100%;
    height: 50%;
    flex-direction: column;
    align-items: center;
    margin: 0 auto;
    position: absolute;
    top: 115px;
    color: white;
}

.movie_item:hover {
    .movie_hover {
        display: flex;
        width: 100%;
        height: 50%;
        flex-direction: column;
        align-items: center;
        margin: 0 auto;
        position: absolute;
        top: 115px;
        color: white;
    }
    .movie_img {
        opacity: 0.3
    }
}


.hover_button_play {
    width: 96px;
    height: 96px;
    background: url('../../image/play96.png') no-repeat center;
    border: 0px;
    margin-top: 20px;
}

.hover_watch {
    margin-top: 20px;
    font-size: 25px;
}

.hover_button_view {
    width: 35%;
    padding: 10px;
    background-color: transparent;
    color: white;
    border: 2px solid white;
    font-size: 20px;
    margin-top: 40px;
}

// ************the end hover movie item****************

.movie_description {
    display: flex;
    justify-content: space-between;
    background-color: white;
    position: absolute;
    top: 455px;
    height: 100px;
    width: 100%;
}

.title {
    margin: 10px 20px 0px 20px;
    color: black;
    text-transform: uppercase;
    height: 30px;
    overflow: hidden;
}

.movie_genre {
    li {
        list-style-type: none;
        display: inline-block;
        margin:10px;
        font-size: 20px;
    }
}

.movie_rating {
    width: 7%;
    height: 24%;
    padding: 3px;
    margin: 40px 60px 0px 0px;; 
    color: #09ade3;
    border: 1px solid #09ade3;;
    font-size: 20px;
    text-align: center;
}

//start discription state//

.discription_hover {
    background-color: rgba(0,0,0,0.5);
    background-repeat: no-repeat;
    background-size: cover;
    height: 550px;
    color: white;
    display: flex;
    flex-direction: column;
    z-index: 10;
    top: 0px;
    .button_close {
        margin: 15px;
        width: 30px;
        height: 30px;
    }
    .discription_title {
        padding: 10px;
        display: flex;
        justify-content: space-between;
    }
}

.button_watch {  
    width: 40%;
    padding: 10px;
    background-color: transparent;
    color: white;
    border: 2px solid white;
    font-size: 20px;
    margin-left: 30%;
}

.main {
    padding: 15px;
    color: white;
}

.img_discription {
    opacity: 0.3;
    height: 550px;
    z-index: 2;
    position: absolute;
}

.button_close {
    background:url('../../image/iconclose.png') no-repeat center;
    border: none;
    padding: 12px 12px;
}

.vote_average {
    border: 2px solid white;
    height: 20px;
    width: 25px;
    padding: 5px;
    margin: 10px 30px;
    text-align: center;
}

.description_img {
    height: 550px;
    width: 100%;
    position: relative;
    text-align: center;
    opacity: 0.3;
}

.discription_item {
    z-index: 1;
    position: relative;
    background: black;
    height: 550px;
}
.description_button {
    background:url('../../image/iconclose.png') no-repeat center;
    border: none;
    padding: 12px 12px;
    top: 10px;
    right: 10px;
    width: 30px;
    height: 30px;
    position: absolute;
    z-index: 1;
}

.description_info {
    position: absolute;
    top: 0px;
    color: white;
}

.description_header {
    display: flex;
    padding: 20px;
}

.description_list {
    list-style-type: none;
    display: inline-flex;
}

.description_list li {
    margin-right: 15px;
}

.discription_overview {
    max-height: 255px;
    overflow: hidden;
}


