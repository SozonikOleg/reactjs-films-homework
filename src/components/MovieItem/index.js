/*eslint-disable */
export { default } from './MovieItem';
