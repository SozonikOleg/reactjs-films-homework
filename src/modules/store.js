import { createStore } from 'redux';

function counter(
  state = {
    oleg: 'First date',
    inna: 'Second date',
  },
  action,
) {
  switch (action.type) {
    case 'INCREMENT':
      return state + 1;
    case 'DECREMENT':
      return state - 1;
    default:
      return state;
  }
}

const store = createStore(counter);

export default store;
