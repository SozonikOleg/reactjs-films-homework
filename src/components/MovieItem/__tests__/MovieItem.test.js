import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import MovieItem from '../MovieItem';
import store from '../../../modules/store';

it('Link changes the class when hovered', () => {
  const component = renderer.create(
    <Provider store={store}>
      <MovieItem />
    </Provider>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
