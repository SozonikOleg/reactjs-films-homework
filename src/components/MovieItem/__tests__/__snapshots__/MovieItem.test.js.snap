// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link changes the class when hovered 1`] = `
<div
  className="movie_wrapper"
>
  <section
    className="movie_item"
  >
    <div
      className="background_img"
    >
      <img
        alt="movie_img"
        className="movie_img"
        src="test-file-stub"
      />
      <div
        className="movie_hover"
      >
        <button
          className="hover_button_play"
          onClick={[Function]}
          type="button"
        />
        <span
          className="hover_watch"
        >
          Watch Now
        </span>
        <button
          className="hover_button_view"
          onClick={[Function]}
          type="button"
        >
          View Info
        </button>
      </div>
    </div>
    <div
      className="movie_description"
    >
      <div
        className="movie-info"
      >
        <h2
          className="title"
        />
        <ul
          className="movie_genre"
        >
          <li>
            Action
          </li>
          <li>
            Drama
          </li>
          <li>
            Comedy
          </li>
        </ul>
      </div>
      <div
        className="movie_rating"
      />
    </div>
  </section>
</div>
`;
